GNUmakefile

```diff
@@ -1,4 +1,13 @@
 all: mentor-matcher
 
-mentor-matcher: main.go
+mentor-matcher: build-binary
+
+build-binary: main.go
```

Comment:
The rules aren't quite right now.
Before the rule stated that the file 'mentor-matcher' could be made from the source file 'main.go' by running the command 'go build'. However that is no longer the case,
instead the recipe for 'mentor-matcher' has a dependency on the file 'build-binary' (which isn't meant to be a file) which itself has a dependency on the file 'main.go'.
Make will always build the file 'mentor-matcher' regardless of the state of 'main.go', since it directly depends upon the file 'build-binary' which never will exist.

My original intention for this recipe was for it to be a build recipe as opposed to a command alias that is run unconditionally.

Reply:
Check out phony targets for command aliases too: https://www.gnu.org/software/make/manual/html_node/Phony-Targets.html